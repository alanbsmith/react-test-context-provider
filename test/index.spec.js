import React from 'react';
import renderer from 'react-test-renderer';
import getElementWithContext from '../lib';

import { TOGGLE_TEST_CONTEXT } from './constants';

import TestSwitch from './TestSwitch';

const context = {
  [TOGGLE_TEST_CONTEXT]: {
    handleToggle: () => {},
  },
};

describe('TestSwitch.Button', () => {
  it('renders correctly', () => {
    const element = getElementWithContext(
      context,
      <TestSwitch>
        <TestSwitch.Button>button text</TestSwitch.Button>
      </TestSwitch>,
    );
    const component = renderer.create(element);
    expect(component).toMatchSnapshot();
  });
});
