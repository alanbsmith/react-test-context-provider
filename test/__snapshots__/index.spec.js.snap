// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSwitch.Button renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    button text
  </button>
</div>
`;
